pallets/loans/src/lib.rs

```diff
@@ -1784,13 +1855,25 @@ impl<T: Config> Pallet<T> {
     // Returns `Err` If InsufficientLiquidity
     // `account`: account that need a liquidity check
     // `reduce_amount`: values that will have an impact on liquidity
-    fn ensure_liquidity(account: &T::AccountId, reduce_amount: FixedU128) -> DispatchResult {
-        let (liquidity, _) = Self::get_account_liquidity(account)?;
-        if liquidity < reduce_amount {
-            Err(Error::<T>::InsufficientLiquidity.into())
-        } else {
-            Ok(())
+    // `dot_lf`: check in liquidation free mode which means borrowing dot or redeeming assets in
+    // `LiquidationFreeCollaterals`.
+    fn ensure_liquidity(
+        account: &T::AccountId,
+        reduce_amount: FixedU128,
+        dot_lf: bool,
+    ) -> DispatchResult {
+        let (total_liquidity, _) = Self::get_account_liquidity(account)?;
+        let lf_liquidity = Self::get_account_lf_liquidity(account)?;
```

Comment:
move this into the if block next line to avoid unnecessary calc

Reply:
yes,realize in our current design we need to check lf_liquidity for none-lf-asset. so just ignore this comment